### PR TITLE
DOC: don't include all methods/attributes of IntervalIndex

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Alternative for #16050 

I totally forgot I already had made such a "class without autosummary table" template before for Categorical, so I hope this should work for IntervalIndex as well.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@TomAugspurger I hope this fixes the warnings.

Maybe we could also use this for the other Index subclasses like DatetimeIndex, MultiIndex, ... although it would be nice to still have a list of those attributes that are special to them, and to refer to the main Index docstring page for a list of all methods. But not sure if such flexibility is possible with sphinx/jinja (and for a later PR in any case :-))

**Comment by jorisvandenbossche** [comment]
Another sphinx / rst mystery ... 
So for some reason this works for Categorical, but not for IntervalIndex. No idea why. 
And the strange thing is that I originally said in the PR that introduced this that "it was not yet working" :-) (https://github.com/pandas-dev/pandas/pull/9721). But apparently is does, at least for Categorical.

**Comment by TomAugspurger** [comment]
Strange... I thought I had tried it too, but I can't remember the outcome. Did you clear all the generated files before rebuilding?

**Comment by TomAugspurger** [comment]
@jorisvandenbossche I wonder if it's https://github.com/pandas-dev/pandas/commit/cf40991093e746e5dde40fca4f322ecfc3dfff78#diff-4501fdb73fd4b9f09d44a3f4177d44aaR46

We special case `.Categorical` there.

**Comment by TomAugspurger** [comment]
Here's what it looks like
screen shot 2017-05-04 at 3 06 10 pm

No new warnings with that.

I spent a bit longer trying to get a an autosummary with a subset of the Methods specified in a `Methods` section, but no luck. Will probably require some modifications to numpydoc (beyond just templates), which will have to wait till after the release.

**Comment by jorisvandenbossche** [comment]
Aha, you have found out how I hacked it before .. :-) Thanks!

File: doc/sphinxext/numpydoc/numpydoc.py [hunk 1]
```diff
@@ -43,7 +43,10 @@ def mangle_docstrings(app, what, name, obj, options, lines,
               )
 
     # PANDAS HACK (to remove the list of methods/attributes for Categorical)
-    if what == "class" and name.endswith(".Categorical"):
+    if what == "class" and (name.endswith(".Categorical") or
+                            name.endswith("CategoricalIndex") or
+                            name.endswith("Interval") or
```
Comment (jorisvandenbossche) [inline doc/sphinxext/numpydoc/numpydoc.py]: @TomAugspurger I think Interval class itself is OK to have with all its methods / attributes ?

Comment (TomAugspurger) [inline doc/sphinxext/numpydoc/numpydoc.py]: Oh, right... Is that in the api.rst?

Comment (TomAugspurger) [inline doc/sphinxext/numpydoc/numpydoc.py]: Doesn't look like it, but I'll remove it anyway.

**Comment by TomAugspurger** [comment]
Doc build log [looks good](https://travis-ci.org/pandas-dev/pandas/jobs/228882935), merging:

**Comment by jorisvandenbossche** [comment]
Great!

**Comment by jreback** [comment]
yep looks great. thanks guys!

**Pull request merged**